azurerm/internal/services/sentinel/sentinel_data_connector_office_365.go

```diff
@@ -0,0 +1,244 @@
+		}
+
+		if err := assertDataConnectorKind(resp.Value, securityinsight.DataConnectorKindOffice365); err != nil {
+			return fmt.Errorf("asserting Sentinel Data Connector of %q: %+v", id, err)
+		}
+		param.Etag = resp.Value.(securityinsight.OfficeDataConnector).Etag
+	}
+
+	_, err = client.CreateOrUpdate(ctx, id.ResourceGroup, operationalInsightsResourceProvider, id.WorkspaceName, id.Name, param)
+	if err != nil {
+		return fmt.Errorf("creating Sentinel Data Connector Office 365 %q: %+v", id, err)
+	}
+
+	d.SetId(id.ID())
+
+	return resourceSentinelDataConnectorOffice365Read(d, meta)
+}
+
+func resourceSentinelDataConnectorOffice365Read(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*clients.Client).Sentinel.DataConnectorsClient
+	ctx, cancel := timeouts.ForRead(meta.(*clients.Client).StopContext, d)
+	defer cancel()
+
+	id, err := parse.DataConnectorID(d.Id())
+	if err != nil {
+		return err
+	}
+	workspaceId := loganalyticsParse.NewLogAnalyticsWorkspaceID(id.SubscriptionId, id.ResourceGroup, id.WorkspaceName)
+
+	resp, err := client.Get(ctx, id.ResourceGroup, operationalInsightsResourceProvider, id.WorkspaceName, id.Name)
+	if err != nil {
+		if utils.ResponseWasNotFound(resp.Response) {
+			log.Printf("[DEBUG] Sentinel Data Connector Office 365 %q was not found - removing from state!", id)
+			d.SetId("")
+			return nil
+		}
+
+		return fmt.Errorf("retrieving Sentinel Data Connector Office 365 %q: %+v", id, err)
+	}
+
+	if err := assertDataConnectorKind(resp.Value, securityinsight.DataConnectorKindOffice365); err != nil {
+		return fmt.Errorf("asserting Sentinel Data Connector Office 365 of %q: %+v", id, err)
+	}
+	dc := resp.Value.(securityinsight.OfficeDataConnector)
+
+	d.Set("name", id.Name)
+	d.Set("log_analytics_workspace_id", workspaceId.ID())
+	d.Set("tenant_id", dc.TenantID)
+
+	if dt := dc.DataTypes; dt != nil {
+		if exchange := dt.Exchange; exchange != nil {
+			d.Set("exchange_enabled", strings.EqualFold(string(exchange.State), string(securityinsight.Enabled)))
+		}
+		if sharePoint := dt.SharePoint; sharePoint != nil {
+			d.Set("share_point_enabled", strings.EqualFold(string(sharePoint.State), string(securityinsight.Enabled)))
+		}
+		if teams := dt.Teams; teams != nil {
+			d.Set("teams_enabled", strings.EqualFold(string(teams.State), string(securityinsight.Enabled)))
+		}
+	}
+
+	return nil
+}
+
+func resourceSentinelDataConnectorOffice365Delete(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*clients.Client).Sentinel.DataConnectorsClient
+	ctx, cancel := timeouts.ForDelete(meta.(*clients.Client).StopContext, d)
+	defer cancel()
+
+	id, err := parse.DataConnectorID(d.Id())
+	if err != nil {
+		return err
+	}
+
+	_, err = client.Delete(ctx, id.ResourceGroup, operationalInsightsResourceProvider, id.WorkspaceName, id.Name)
+	if err != nil {
+		return fmt.Errorf("deleting Sentinel Data Connector Office 365 %q: %+v", id, err)
```

Comment:
we can reduce this to:

```suggestion
		return fmt.Errorf("deleting %s: %+v", id, err)
```